dev/tasks/verify-rc/github.macos.amd64.yml

```diff
@@ -31,20 +33,36 @@ jobs:
     {% endif %}
 
     steps:
-      {{ macros.github_checkout_arrow()|indent }}
+      {{ macros.github_checkout_arrow(fetch_depth=0)|indent }}
 
+      {% if not use_conda %}
       - name: Install System Dependencies
         shell: bash
         run: |
           brew update
           brew bundle --file=arrow/cpp/Brewfile
           brew bundle --file=arrow/c_glib/Brewfile
+      {% endif %}
+
+      - uses: actions/setup-java@v2
+        with:
+          distribution: 'temurin'
+          java-version: '11'
+
       - uses: actions/setup-node@v2-beta
         with:
           node-version: '16'
+
       - name: Run verification
         shell: bash
+        env:
+          TEST_DEFAULT: 0
+        {% if target is defined %}
```

Comment:
Going to remove it.

Reply:
`target` was undefined for binary, wheels and jars verification tasks. I ended up removing the `artifact` argument of the script in favor or environment flags.

So instead of 

```
dev/release/verify-release-candidate.sh source <version> <rc>
```

```
TEST_DEFAULT=0 TEST_SOURCE=1 dev/release/verify-release-candidate.sh <version> <rc>
```